src/main/java/io/jenkins/plugins/pipelinegraphview/consoleview/PipelineConsoleViewAction.java

```diff
@@ -120,40 +121,23 @@ protected JSONObject getAllSteps() throws IOException {
     }
 
     @WebMethod(name = "log")
-    public HttpResponse getConsoleText(StaplerRequest2 req, StaplerResponse2 rsp) throws IOException {
+    public void getConsoleText(StaplerRequest2 req, StaplerResponse2 rsp) throws IOException {
         String nodeId = req.getParameter("nodeId");
         if (nodeId == null) {
             logger.error("'consoleText' was not passed 'nodeId'.");
-            return HttpResponses.errorJSON("Error getting console text");
+            rsp.getWriter().write("Error getting console text");
+            return;
         }
         logger.debug("getConsoleText was passed node id '{}'.", nodeId);
         // This will be a step, so return its log output.
         AnnotatedLargeText<? extends FlowNode> logText = getLogForNode(nodeId);
 
-        long count = 0;
-        PipelineStepList steps = stepApi.getSteps(nodeId);
-        try (CharSpool spool = new CharSpool()) {
-
-            for (PipelineStep step : steps.getSteps()) {
-                AnnotatedLargeText<? extends FlowNode> logForNode = getLogForNode(String.valueOf(step.getId()));
-                if (logForNode != null) {
-                    count += logForNode.writeLogTo(0, spool);
-                }
-            }
-
-            if (count > 0) {
-                rsp.setContentType("text/plain;charset=UTF-8");
-                try (Writer writer = rsp.getWriter()) {
-                    spool.flush();
-                    spool.writeTo(new LineEndNormalizingWriter(writer));
-                }
-            }
-        }
 
         if (logText != null) {
-            return HttpResponses.text(PipelineNodeUtil.convertLogToString(logText));
+            logText.writeLogTo(0L, rsp.getOutputStream());
```

Comment:
Note that the return value is, in general, significant.

Reply:
Is it when writing to an http output stream?

> if the file is still being written, this method writes the file until the last newline character and returns the offset to start the next write operation.

does that mean that you need to loop until no offset? (the plugin doesn't currently do this and I'm not aware of any issues occurring currently)

Reply:
Probably rarely matters.